### ISSUE TITLE
JWT Authentication Middleware

### ISSUE THREAD

**Issue by adrian0007** [jupitern/slim3-skeleton#6]
Hi, I'm trying to implement [Tuupola's JWT Auth](https://github.com/tuupola/slim-jwt-auth) middleware in this awesome project, so I added to `config/app.php`:
```php
'middleware' => [
   //App\Middleware\Session::class,
       Tuupola\Middleware\JwtAuthentication::class,
],
```
However, the `public static registerMiddleware()` does not read the [] parameters which are required by the middleware:
```php
$app->add(new \Tuupola\Middleware\JwtAuthentication([
    "secret" => "supersecretkeyyoushouldnotcommittogithub",
    "path" => "/api",
    "ignore" => ["/api/token", "/api/test"],
    "algorithm" => ["HS256"],
    "attribute" => "jwt",
    "callback" => function ($request, $response, $arguments) use ($container) {
        $container["token"] = $arguments["decoded"];
    },
    "error" => function ($response, $arguments) {
        $data["status"] = "error";
        $data["message"] = $arguments["message"];
        return $response
            ->withHeader("Content-Type", "application/json")
            ->getBody()->write(json_encode($data, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT));
    }
]));
```
I tried to add the above directly into `bootstrap.php`, but the variable in `use ($container)` is undefined in this context. In another [example ](https://github.com/tuupola/slim-api-skeleton/blob/9324b95fc05e72cdc299be75389b5cb817e71799/config/middleware.php#L40) from Tuupola, a `container['token']` was defined, what could be the equivalent of that in this project (perhaps the constructor of `lib/framework/app.php` ?)

I also looked at the `Session` example middleware included in this skeleton which uses the `__invoke(..)` magic method, but I was not very successful in converting the JWT Authentication to this format either.

Can someone help me with a fresh pair of eyes please?

**Comment by jupitern** [comment]
Hi,
You can add a service provider instead and register it in "config/app.php" or you can add you middleware with params in bootstrap.php after $app->registerMiddleware();

Service provider file "app\ServiceProviders\JwtAuth.php":
````
<?php

namespace App\ServiceProviders;
use Tuupola\Middleware\JwtAuthentication;


class JwtAuth implements ProviderInterface
{

    public static function register()
    {
        app()->add(new JwtAuthentication([
            "secret" => "supersecretkeyyoushouldnotcommittogithub",
        ]));
    }

}

**Comment by adrian0007** [comment]
Thank you very much, I'm really enjoying what you've done so far 👍 

I noticed that "resolveRoute" does not play well with arrays ( i.e. `return ['abc' => 'def']` ) and I have to wrap the array it in json_encode before it completes its journey back into routes.php ([this line](https://github.com/jupitern/slim3-skeleton/blob/8a79992c8044d2347df7419b89674f62a2ec7f5a/lib/Framework/App.php#L203) returns an empty object).. am i missing something?

**Comment by jupitern** [comment]
i think you found a bug. The reponse object is immutable for in that line i have to do:
````
$response = $response->withJson($resp);

**Comment by jupitern** [comment]
fixed in version 2.2.1
